src/tree-manager/index.js

```diff
@@ -85,29 +85,37 @@ class TreeManager {
 
   /**
   * If the node didn't specify anything on its own
-  * figure out the initial state based on parent selections
+  * figure out the initial state for checked and disabled based on parent
   * @param {object} node [description]
   */
-  setInitialCheckState (node) {
-    if (node.checked === undefined) node.checked = this.getNodeCheckedState(node)
+  setInitialStatus (node) {
+    if (node.checked === undefined || node.disabled === undefined) {
```

Comment:
I think this check should not be done. If either `checked` or `disabled` is set, it'll skip assignment for both props. I guess just having:

```
const { parentCheckState, parentDisabledState } = this.getNodeStatus(node)
if (node.checked === undefined) node.checked = node.checked || parentCheckState
if (node.disabled === undefined) node.disabled = node.disabled || parentDisabledState
```

would suffice (and work correctly even if one is set and the other is not).